apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -90,6 +91,20 @@ function renderSectionProgressReact(store) {
   );
 }
 
+$(document).ready(function () {
+  const gdprData = scriptData.gdpr;
+  const studioUrlPrefix = scriptData.studiourlprefix;
+  ReactDOM.render(
+    <GDPRDialog
+      isDialogOpen={gdprData.show_gdpr_dialog}
+      currentUserId={gdprData.current_user_id}
+      requestIp={gdprData.request_ip}
+      studioUrlPrefix={studioUrlPrefix}
+    />,
+    document.getElementById('gdpr-dialog')
```

Comment:
Nit: This would be slightly better encapsulated by letting your JS generate the DOM container for React instead of depending on a particular element being previously rendered by the HAML view - that cuts the responsibility of the HAML down to rendering the correct script and scriptdata.  [Example here.](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/ui/accounts/ChangeEmailController.js#L40-L69)